### PR TITLE
Multi-page PDF scanner

### LINKED ISSUE #10167: Scan document: create PDF
### Steps to reproduce
1. Update to Android Nextcloud app, version 3.20.0
2. Navigate to destination folder, tap +, choose scan document from camera
3. Take picture of document, follow prompts to finish
4. Nextcloud file list updates, new file appears in JPG format

### Expected behaviour
- Tell us what should happen
- File should upload in PDF format

### Actual behaviour
- Tell us what happens
- File uploads in JPG format

### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.
- If yes, please open up a bug report
- If not, please verify server setup and ask for help on forum

### Environment data
Android version: 10

Device model: Samsung Galaxy S9

Stock or customized system: Stock

Nextcloud app version: 3.20.0

Nextcloud server version: 23.0.3

Reverse proxy:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

### LINKED ISSUE #11150: Scan Multiple Documents at once
Currently we can only scan one paper at a time. It would be nice to be able to scan multiple pages at once and combine them into one file. I think for this to happen this would need to be implemented https://github.com/nextcloud/android/issues/10167

### PR DESCRIPTION
**Pull request by AlvaroBrey**
Includes a refactor to make the optional feature a separate module, avoiding code duplication between variants as much as possible.

- [x] Request multiple pages from scanner
- [x] Preview scanned pages
- [x] Upload those pages at the same time
- [x] Generate PDF from images
- [x] Choice between PDF (single doc) or JPEG (multiple pages)
- [x] Tests written, or not not needed
    - Might write more depending on time

Fixes #11150
Fixes #10167

Related: #11021, although not fixed completely for now

### CODE REVIEW

**Comment by AlvaroBrey** [comment]
Might write more tests depending on time

**Comment by tobiasKaminsky** [comment]
Works really well on first try :clap: :clap: :clap: 
I noticed only
- click + to add document -> click cancel -> you see white overview to capture an image. I think it is best if cancel closes this if there is none image taken

File: app/src/generic/java/com/nextcloud/client/di/VariantModule.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Nextcloud Android client application
+ *
+ * @author Chris Narkiewicz
+ * Copyright (C) 2919 Chris Narkiewicz <hello@ezaquarii.com>
```
Comment (tobiasKaminsky) [inline app/src/generic/java/com/nextcloud/client/di/VariantModule.kt:5]: Name?

File: app/src/main/res/drawable/ic_save.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<!-- drawable/content_save.xml -->
```
Comment (tobiasKaminsky) [inline app/src/main/res/drawable/ic_save.xml:1]: License?

**Comment by tobiasKaminsky** [comment]
Do we need
- appscan/consumer-rules.pro
- appscan/proguard-rules.pro

File: app/src/main/res/drawable/ic_save.xml (hunk 2 above)
Comment (AlvaroBrey) [reply app/src/main/res/drawable/ic_save.xml:1]: It's from MDI which has Apache license, but it doesn't have a header. Not sure what to do about that

Comment (AlvaroBrey) [reply app/src/main/res/drawable/ic_save.xml:1]: > We have made these icons available for you to incorporate them into your products under the [Apache License Version 2.0](http://www.apache.org/licenses/LICENSE-2.0.txt). Feel free to remix and re-share these icons and documentation in your products. We'd love attribution in your app's about screen, but it's not required. The only thing we ask is that you not re-sell the icons themselves.

Comment (AlvaroBrey) [reply app/src/main/res/drawable/ic_save.xml:1]: So yeah, no action needed

**Comment by AlvaroBrey** [comment]
> Do we need
> 
> * appscan/consumer-rules.pro
> * appscan/proguard-rules.pro

Nope. Removed.

**Comment by AlvaroBrey** [comment]
> * click + to add document -> click cancel -> you see white overview to capture an image. I think it is best if cancel closes this if there is none image taken
> * start adding images -> rotate device -> camera is gone
> * have no camera permission -> start document scanning -> deny camera request -> should return to list view again

All fixed

**Comment by tobiasKaminsky** [comment]
Awesome work!

**Review by tobiasKaminsky — APPROVED** [review]

**Pull request merged**